### PR TITLE
Fpp

### PR DESCRIPTION
**Pull request by mateusz**

### CODE REVIEW

File: mysql_mutex.go [hunk 1]
```diff
@@ -2,37 +2,50 @@ package main
 
 import (
 	"database/sql"
-	"errors"
 	"fmt"
-	_ "github.com/go-sql-driver/mysql"
 	"sync"
 	"time"
+
```
Comment (halkyon) [inline mysql_mutex.go:8]: why the newline?

File: mysql_mutex_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+func (r *respMock) Close() error {
+	return nil
+}
+func (r *respMock) Next() bool {
+	haveSome := r.rowsAvailable > 0
+	r.rowsAvailable--
+	return haveSome
+}
+func (r *respMock) Scan(dest ...interface{}) error {
+	fmt.Printf("%#v\n", dest)
```
Comment (stojg) [inline mysql_mutex_test.go]: can be dropped?

File: mysql_mutex_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	*(dest[0].(*int)) = r.mockValue
+	fmt.Printf("%#v\n", dest)
```
Comment (stojg) [inline mysql_mutex_test.go]: can be dropped?

File: sera.go [hunk 4]
```diff
@@ -91,24 +98,33 @@ func realMain() (int, error) {
...
 	defer db.Close()
 
-	mutex := NewMysqlMutex(db, keyName, timeout)
+	mutex := NewMysqlMutex(&MysqlConnection{DB: db}, keyName, timeout)
 
 	// Try to get the lock, block until we get the lock or we reached the timeout value
 	if err = mutex.Lock(); err != nil {
+		// In first past the post mode, we are happy for subsequent agents to not execute.
+		if _, ok := err.(ErrLockTaken); ok && conf.IgnoreUnacquirable {
+			fmt.Println("Unable to acquire lock, but reporting success.")
```
Comment (stojg) [inline sera.go]: I would check for conf.Verbose, otherwise it would cause a lot of noise in logs etc for an expected behaviour.

File: sera.go [hunk 5]
```diff
@@ -41,9 +42,15 @@ const (
 
 // Represents the json config for sera
 type Config struct {
-	Server  string `json:"server"`
-	Syslog  bool   `json:"syslog"`
-	Verbose bool   `json:"verbose"`
+	Server             string `json:"server"`
+	Syslog             bool   `json:"syslog"`
+	Verbose            bool   `json:"verbose"`
+	IgnoreUnacquirable bool   `json:"-"`
```
Comment (stojg) [inline sera.go]: you can just make this unexported and it wouldn't be picked up the JSON marshaller

File: mysql_mutex.go (hunk 1 above)
Comment (mateusz) [reply mysql_mutex.go:8]: vim-go added it :-D

File: mysql_mutex_test.go (hunk 2 above)
Comment (mateusz) [reply mysql_mutex_test.go]: 👍

File: mysql_mutex_test.go (hunk 3 above)
Comment (mateusz) [reply mysql_mutex_test.go]: 👍

File: sera.go (hunk 4 above)
Comment (mateusz) [reply sera.go]: 👍

File: sera.go (hunk 5 above)
Comment (mateusz) [reply sera.go]: 👍

**Comment by stojg** [comment]
suggestion, change the flag name to something more human brain parsable, like `--wait-and-skip`

**Pull request merged**